test/e2e/dnslookup_test.go

```diff
@@ -130,20 +133,32 @@ func TestMain(m *testing.M) {
 	err = writeDNSHammerFile(filepath.Join(tmpDir, dnsHammerCNAMERecords), cnameRecords)
 	requireNoErrorf(err, "Failed generating CNAME test data")
 
-	logrus.Infof("Dowloading %v wsl distro tarball", wslTarballName)
-	tarballPath := filepath.Join(tmpDir, wslTarballName)
-
-	err = downloadFile(tarballPath, wslTarballURL)
-	requireNoErrorf(err, "Failed to download wsl distro tarball: %v", err)
+	tarballPath := filepath.Join(os.TempDir(), wslTarballName)
```

Comment:
Why don't you use `tmpdir` here?

```suggestion
	tarballPath := filepath.Join(tmpdir, wslTarballName)
```

Reply:
This is intentional, `tmpdir` is only available within the life of each test run and it gets removed after each run, however,  `os.TmpDir()` is bound to the user's OS session, which makes it more ideal location for caching.

Reply:
You are literally setting `tmpDir = os.TempDir()` just 20 lines earlier; how can this be a different value now?

Reply:
Just noticed I wrote `tmpdir` instead of `tmpDir` in the suggestion above; I meant the latter. But there is no `tmpdir` in the code, so I think this was still clear.

Reply:
I was under the assumption that host-resovler e2e is doing the same thing as Vtunnel [here](https://github.com/rancher-sandbox/rancher-desktop/blob/1e31560b06c62da5333d7c4f5e75ea86d1915ba3/src/go/vtunnel/test/e2e/connectivity_test.go#L255) since I was working on them in parallel, I will update.

Reply:
From discussing with Nino offline: this is intentional so that we _don't_ delete the tarball (so it can be reused on next run).